xtask/src/run.rs

```diff
@@ -1,47 +1,66 @@
 use std::{
+    env::consts::{ARCH, OS},
     ffi::OsString,
     fmt::Write as _,
-    io::BufReader,
-    path::PathBuf,
-    process::{Child, Command, Stdio},
+    fs::{copy, create_dir_all, metadata, File},
+    io::{BufRead as _, BufReader, ErrorKind, Write as _},
+    path::{Path, PathBuf},
+    process::{Child, Command, Output, Stdio},
 };
 
 use anyhow::{anyhow, bail, Context as _, Result};
 use cargo_metadata::{Artifact, CompilerMessage, Message, Target};
 use clap::Parser;
-use xtask::AYA_BUILD_INTEGRATION_BPF;
+use xtask::{exec, AYA_BUILD_INTEGRATION_BPF};
 
-#[derive(Debug, Parser)]
-pub struct BuildOptions {
-    /// Arguments to pass to `cargo build`.
-    #[clap(long)]
-    pub cargo_arg: Vec<OsString>,
+#[derive(Parser)]
+pub enum Environment {
+    /// Runs the integration tests locally.
+    Local {
+        /// The command used to wrap your application.
+        #[clap(short, long, default_value = "sudo -E")]
+        runner: String,
+    },
+    /// Runs the integration tests in a VM.
+    VM {
+        /// The kernel images to use.
+        ///
+        /// You can download some images with:
+        ///
+        /// wget --accept-regex '.*/linux-image-[0-9\.-]+-cloud-.*-unsigned*' \
+        ///   --recursive ftp://ftp.us.debian.org/debian/pool/main/l/linux/
+        ///
+        /// You can then extract them with:
+        ///
+        /// find . -name '*.deb' -print0 \
+        /// | xargs -0 -I {} sh -c "dpkg --fsys-tarfile {} \
+        ///   | tar --wildcards --extract '*vmlinuz*' --file -"
+        #[clap(required = true)]
+        kernel_image: Vec<PathBuf>,
```

Comment:
Is it useful to pass > 1 kernel? In CI I'd assume we'd use a matrix job.
For local testing, I'm not sure...